crates/polars-core/src/chunked_array/logical/datetime.rs

```diff
@@ -81,21 +81,39 @@ impl LogicalType for DatetimeChunked {
             },
             #[cfg(feature = "dtype-time")]
             (Datetime(tu, _), Time) => match tu {
-                TimeUnit::Nanoseconds => Ok((self.0.as_ref() % NS_IN_DAY)
-                    .cast(&Int64)
-                    .unwrap()
-                    .into_time()
-                    .into_series()),
-                TimeUnit::Microseconds => Ok((self.0.as_ref() % US_IN_DAY * 1_000i64)
-                    .cast(&Int64)
-                    .unwrap()
-                    .into_time()
-                    .into_series()),
-                TimeUnit::Milliseconds => Ok((self.0.as_ref() % MS_IN_DAY * 1_000_000i64)
-                    .cast(&Int64)
-                    .unwrap()
-                    .into_time()
-                    .into_series()),
+                TimeUnit::Nanoseconds => {
+                    let ca = self.0.apply_values(|v| {
```

Comment:
I timed this separately, and `apply_values` is faster there than `apply`

Reply:
This is a bit more concise (I'm always looking to reduce code length) and possibly faster since it removes the `if` branch (although maybe a premature optimization):

```rust
            #[cfg(feature = "dtype-time")]
            (Datetime(tu, _), Time) => Ok({
                let (modder, multiplier) = match tu {
                    TimeUnit::Nanoseconds => (NS_IN_DAY, 1i64),
                    TimeUnit::Microseconds => (US_IN_DAY, 1_000i64),
                    TimeUnit::Milliseconds => (MS_IN_DAY, 1_000_000i64),
                };
                self.0
                    .apply_values(|v| (v % modder * multiplier) + (NS_IN_DAY * (v < 0) as i64))
                    .cast(&Int64)
                    .unwrap()
                    .into_time()
                    .into_series()
            }),
```

Reply:
I thinks this is a nice improvement. @mcrumiller :+1:

Reply:
yup - @mcrumiller fancy opening a PR to my PR, so you're included in the commit?

Reply:
Sure (although I don't care about the cred)--need to figure out how to check out a branch on another fork first.

Reply:
good learning experience then :) but yes, that's how I'd have done it

alternatively,
```
git checkout -b pr/marcogorelli/man-on-the-moon
git fetch marcogorelli
git reset --hard marcogorelli/man-on-the-moon
<do your changes>
git push -U marcogorelli HEAD:man-on-the-moon
```

Reply:
Got that working, now trying to figure out how to do create the PR, as your fork doesn't seem to show up in git's list:

![image](https://github.com/pola-rs/polars/assets/1896992/e53e78ca-85ad-4807-a834-5041caa52318)

Reply:
Edit: w00t